editor/editor_main_screen.cpp

```diff
@@ -233,6 +233,24 @@ EditorPlugin *EditorMainScreen::get_plugin_by_name(const String &p_plugin_name)
 	return main_editor_plugins[p_plugin_name];
 }
 
+bool EditorMainScreen::can_auto_switch_screens() const {
+	if (selected_plugin == nullptr) {
+		return true;
+	}
+	Node *script_tab = button_hb->get_node(NodePath("Script"));
+	if (script_tab == nullptr) {
+		return true;
+	}
+	// Only allow auto-switching if the selected tab is to the left of the script tab.
+	for (int i = 0; i < script_tab->get_index(); i++) {
+		Button *button = Object::cast_to<Button>(button_hb->get_child(i));
+		if (button->get_text() == selected_plugin->get_plugin_name()) {
+			return true;
+		}
+	}
+	return false;
```

Comment:
```suggestion
	// Only allow auto-switching if the selected tab is to the left of the script tab.
	for (int i = 0; i < button_hb->get_child_count(); i++) {
		Button *button = Object::cast_to<Button>(button_hb->get_child(i));
		if (button->get_text() == "Script") {
			// Selected button is at or after the Script button.
			return false;
		}
		if (button->get_text() == selected_plugin->get_plugin_name()) {
			// Selected button is before the Script button.
			return true;
		}
	}
	return false;
```

It's better, but I still think it can be simplified further.